data/resources/ui/login.ui

```diff
@@ -2,402 +2,425 @@
 <interface>
   <template class="Login" parent="AdwBin">
     <child>
-      <object class="GtkBox">
-        <property name="orientation">vertical</property>
+      <object class="GtkStack" id="main_stack">
+        <property name="transition-type">crossfade</property>
         <child>
-          <object class="GtkHeaderBar">
-            <child type="start">
-              <object class="GtkButton" id="previous_button">
-                <property name="visible">False</property>
-                <property name="action-name">login.previous</property>
-                <property name="use-underline">True</property>
-                <property name="label" translatable="yes">_Previous</property>
-              </object>
-            </child>
-            <child type="end">
-              <object class="GtkButton">
-                <property name="action-name">login.next</property>
-                <property name="child">
-                  <object class="GtkStack" id="next_stack">
-                    <child>
-                      <object class="GtkLabel" id="next_label">
-                        <property name="use-underline">True</property>
-                        <property name="label" translatable="yes">_Next</property>
-                      </object>
-                    </child>
-                    <child>
-                      <object class="GtkSpinner" id="next_spinner">
-                        <property name="spinning">True</property>
-                        <property name="valign">center</property>
-                        <property name="halign">center</property>
-                      </object>
-                    </child>
-                  </object>
+          <object class="GtkStackPage">
+            <property name="name">flash-prevention-page</property>
```

Comment:
This name really feels that this widget represents an hack, but I don't think it really is, so I suggest naming it something like "loading-page" or even "empty-page".

Reply:
Indeed, I was a too pessimistic here. I changed it to "empty-page" because I think "loading-page" would arouse the association that something is spinning in there.